esp/services/ws_workunits/ws_wudetails.cpp

```diff
@@ -0,0 +1,349 @@
+                                                    IConstWUAttributeToReturn & attribsToReturn)
+{
+    Owned<StatisticsFilter> wustatsfilter = new StatisticsFilter;
+
+    wustatsfilter->setScopeDepth(0, filter.getMaxDepth());
+
+    StatisticMeasure measure = queryMeasure(attribsToReturn.getMeasure());
+    if (measure != SMeasureNone && measure != SMeasureAll)
+        wustatsfilter->setMeasure(measure);
+
+    return wustatsfilter.getClear();
+}
+
+void CWUDetails::buildAttribFilter(IArrayOf<CAttributeFilter> & processedAttribFilter,
+                                   IArrayOf<IConstWUAttributeFilter> & reqAttribFilter)
+{
+    ForEachItemIn(idx,reqAttribFilter)
+    {
+          IConstWUAttributeFilter & attr = reqAttribFilter.item(idx);
+          const char * sStatisticKind = attr.getName();
+          if ( *sStatisticKind==0 ) continue; // skip nulls
+
+          StatisticKind statisticKind = queryStatisticKind(sStatisticKind);
+          if (statisticKind == StKindNone || statisticKind== StKindAll)
+              throw MakeStringException(ECLWATCH_INVALID_INPUT,
+                                        "Invalid Attribute Name ('%s') in Attribute Filter",
+                                        reqAttribFilter.item(idx).getName());
+
+          const char * sExactValue = attr.getExactValue();
+          const char * sMinValue = attr.getMinValue();
+          const char * sMaxValue = attr.getMaxValue();
+          bool bExactValue = *sExactValue!=0;
+          bool bMinValue = *sMinValue!=0;
+          bool bMaxValue = *sMaxValue!=0;
+
+          if (bExactValue && (bMinValue||bMaxValue))
+              throw MakeStringException(ECLWATCH_INVALID_INPUT,
+                                        "Invalid Attribute Filter ('%s') - ExactValue may not be used with MinValue or MaxValue",
+                                        reqAttribFilter.item(idx).getName());
+
+          Owned<CAttributeFilter> caf = new CAttributeFilter(statisticKind, atoi64(sExactValue), atoi64(sMinValue),
+                                                             atoi64(sMaxValue), bExactValue, bMinValue, bMaxValue);
+          processedAttribFilter.append(*(caf.getLink()));
+    }
+    processedAttribFilter.sort(funcCompareAttribFilter);
+}
+
+void CWUDetails::buildAttribListToReturn(UnsignedShortArray & skReturnAttribList,
+                                         StringArray & returnAttriblist)
+{
+    bool isNew;
+    ForEachItemIn(idx,returnAttriblist)
+    {
+        const char * attribName = returnAttriblist[idx];
+        if (*attribName==0) continue;
+
+        StatisticKind sk = queryStatisticKind(attribName);
+        if (sk==StKindNone || sk==StKindAll)
+            throw MakeStringException(ECLWATCH_INVALID_INPUT, "Invalid Attribute name in AttributeToReturn(%s)",attribName);
+
+        unsigned short usk = static_cast<unsigned short> (sk);
+        skReturnAttribList.bAdd(usk, funcCompareUnsignedShort, isNew);
```

Comment:
Was considering making use of the returned isNew to make sure that duplicate filter on the same attibute name is not provided.  For example, if the the request has two attribute filters for "RecordCount", then only one of those filters will be processed - the other will be skipped.  Was about to make use of the isNew to make sure that the attribute filter list doesn't contain duplicates.